### ISSUE TITLE
Changing items does not trigger updating filtered list items

### ISSUE THREAD

**Issue by atomcat1978** [pstanoev/simple-svelte-autocomplete#144]
Updating items invoked prepareListItems  (` $: items, prepareListItems()`), however the 
list of `filteredListItems` stays the same. This leads to inconsistent data view if the dropdown is opened again using the down arrow.

**Comment by atomcat1978** [comment]
I bet, the list should be re-filtered. Something what ` resetListToAllItemsAndOpen` does.

**Comment by atomcat1978** [comment]
Seems like a call to `search` does the trick:
```
  function prepareListItems() {
    let timerId
    if (debug) {
      timerId = `Autocomplete prepare list ${inputId ? `(id: ${inputId})` : ""}`
      console.time(timerId)
      console.log("Prepare items to search")
      console.log("items: " + JSON.stringify(items))
    }

    if (!Array.isArray(items)) {
      console.warn("Autocomplete items / search function did not return array but", items)
      items = []
    }

    const length = items ? items.length : 0
    listItems = new Array(length)

    if (length > 0) {
      items.forEach((item, i) => {
        const listItem = getListItem(item)
        if (listItem === undefined) {
          console.log("Undefined item for: ", item)
        }
        listItems[i] = listItem
      })
      search() // << -- Here
    }

    if (debug) {
      console.log(listItems.length + " items to search")
      console.timeEnd(timerId)
    }
  }
```

**Comment by pstanoev** [comment]
Thanks for reporting, I will have a look.

**Comment by pstanoev** [comment]
Should be fixed in 2.3.5.
